web/packages/teleport/src/AppLauncher/AppLauncher.test.tsx

```diff
@@ -97,54 +101,214 @@ describe('app launcher path is properly formed', () => {
+    expectedPublicAddr: 'test-app.test.teleport',
+    expectedArn: undefined,
+  },
+  {
+    name: 'uppercase public addr, FQDN',
+    path: 'TEST-APP.test.teleport/test.teleport/TEST-APP.test.teleport?state=ABC',
+    returnedFqdn: 'test-app.test.teleport',
+    expectedFqdn: 'test-app.test.teleport',
+    expectedPublicAddr: 'TEST-APP.test.teleport',
+    expectedArn: undefined,
+  },
+  {
+    name: 'uppercase FQDN, resolved FQDN, public addr',
+    path: 'TEST-APP.test.teleport/test.teleport/TEST-APP.test.teleport?state=ABC',
+    returnedFqdn: 'TEST-APP.test.teleport',
+    expectedFqdn: 'test-app.test.teleport',
+    expectedPublicAddr: 'TEST-APP.test.teleport',
+    expectedArn: undefined,
+  },
+  {
+    name: 'public addr with port',
+    path: 'test-app.test.teleport/test.teleport/test-app.test.teleport:443?state=ABC',
+    returnedFqdn: 'test-app.test.teleport',
+    expectedFqdn: 'test-app.test.teleport',
+    expectedPublicAddr: 'test-app.test.teleport',
+    expectedArn: undefined,
+  },
+  {
+    name: 'FQDN with port',
+    path: 'test-app.test.teleport:443/test.teleport/test-app.test.teleport?state=ABC',
+    returnedFqdn: 'test-app.test.teleport',
+    expectedFqdn: 'test-app.test.teleport:443',
+    expectedPublicAddr: 'test-app.test.teleport',
+    expectedArn: undefined,
+  },
+  {
+    name: 'resolved FQDN with port',
+    path: 'test-app.test.teleport/test.teleport/test-app.test.teleport?state=ABC',
+    returnedFqdn: 'test-app.test.teleport:443',
+    expectedFqdn: 'test-app.test.teleport',
+    expectedPublicAddr: 'test-app.test.teleport',
+    expectedArn: undefined,
+  },
+  {
+    name: 'FQDN, public addr with port',
+    path: 'test-app.test.teleport:443/test.teleport/test-app.test.teleport:443?state=ABC',
+    returnedFqdn: 'test-app.test.teleport',
+    expectedFqdn: 'test-app.test.teleport:443',
+    expectedPublicAddr: 'test-app.test.teleport',
+    expectedArn: undefined,
+  },
+  {
+    name: 'FQDN, resolved FQDN with port',
+    path: 'test-app.test.teleport:443/test.teleport/test-app.test.teleport?state=ABC',
+    returnedFqdn: 'test-app.test.teleport:443',
+    expectedFqdn: 'test-app.test.teleport:443',
+    expectedPublicAddr: 'test-app.test.teleport',
+    expectedArn: undefined,
+  },
+  {
+    name: 'public addr, resolved FQDN with port',
+    path: 'test-app.test.teleport/test.teleport/test-app.test.teleport:443?state=ABC',
+    returnedFqdn: 'test-app.test.teleport:443',
+    expectedFqdn: 'test-app.test.teleport',
+    expectedPublicAddr: 'test-app.test.teleport',
+    expectedArn: undefined,
+  },
+  {
+    name: 'FQDN, public addr, resolved FQDN with port',
+    path: 'test-app.test.teleport:443/test.teleport/test-app.test.teleport:443?state=ABC',
+    returnedFqdn: 'test-app.test.teleport:443',
+    expectedFqdn: 'test-app.test.teleport:443',
+    expectedPublicAddr: 'test-app.test.teleport',
+    expectedArn: undefined,
+  },
+];
+
+describe('app session request is properly formed', () => {
```

Comment:
i don't think this test is testing what i think you might be trying to test 😅 

you have test cases trying to test with varying uppercases, so you'd probably want the test to fail if the fqdn matching failed, but we aren't testing for the "success" state of render which means any errors will be silent. to test the success state of render:

calling window.replace for redirecting user (we only redirect if there were no errors) and making sure an error wasn't rendered
```js
      await waitFor(() =>
        expect(window.location.replace).toHaveBeenCalled()
      );

     // for any error, we show a "Access Denied" error card
     expect(screen.queryByText(/access denied/i)).not.toBeInTheDocument()
```

we should also add test for error cases. I would specifically test
- non matching fqdns
- invalid url (put some random chars in the fqdn and that should make the `new URL()` throw)